src/components/content/timeline/patient-timeline.tsx

```diff
@@ -1,29 +1,37 @@
 import cx from "classnames";
+import { useEffect } from "react";
 import { patientTimelineColumns } from "./patient-timeline-columns";
 import { usePatientEncounterDetailsDrawer } from "../timeline-2.0/helpers/modal-hooks";
 import { useCTW } from "@/components/core/providers/ctw-provider";
 import { Table } from "@/components/core/table/table";
 import { usePatientEncounters } from "@/fhir/encounters";
 import { orderBy } from "@/utils/nodash";
+import { Telemetry } from "@/utils/telemetry";
 
 export type PatientTimelineProps = {
   className?: cx.Argument;
 };
 
 export function PatientTimeline({ className }: PatientTimelineProps) {
-  const patientEncounterQuery = usePatientEncounters();
+  const query = usePatientEncounters();
   const { featureFlags } = useCTW();
   const openDetails = usePatientEncounterDetailsDrawer();
   const encounters = orderBy(
-    patientEncounterQuery.data ?? [],
+    query.data ?? [],
     [(encounter) => encounter.resource.period?.start ?? ""],
     ["desc"]
   );
 
+  useEffect(() => {
+    if (!query.isLoading) {
+      Telemetry.reportZAPRecordCount("encounters", query.data?.length);
+    }
+  }, [query.isLoading, query.data]);
```

Comment:
same here (checking query.data exists before calling reportZAPRecordCount

```suggestion
  useEffect(() => {
    if (!query.isLoading && query.data) {
      Telemetry.reportZAPRecordCount("encounters", query.data.length);
    }
  }, [query.isLoading, query.data]);
```